src/commands/init.ts

```diff
@@ -169,40 +174,59 @@ export default class Init extends Command {
+    const scriptsAlreadyThere = untouchedPackageJson.scripts;
 
-    const conflicts = packageJson.scripts ? Object.keys(scripts).filter(name => Object.keys(packageJson.scripts).includes(name)) : [];
+    const conflicts = scriptsAlreadyThere ? Object.keys(scriptsToBeAdded).filter(name => Object.keys(scriptsAlreadyThere).includes(name)) : [];
 
     if (
       conflicts.length &&
       !(await cli.confirm(`Script(s) with the name(s) "${conflicts.join('", "')}" already exist(s). Should they be overwritten?`))
     ) {
-      this.error('Script exits as npm scripts could not be written.', {
+      this.error('Script exits as npm scriptsToBeAdded could not be written.', {
         exit: 11
       });
     }
-    packageJson.scripts = { ...packageJson.scripts, ...scripts };
 
-    if (initializationType === InitType.freshExpress) {
-      packageJson.devDependencies = { ...packageJson.devDependencies, ...packageJsonData.devDependencies };
-    }
+    const adjustedPackageJson = {
+      ...this.packageJson(flags),
+      scripts: { ...untouchedPackageJson.scripts, ...scriptsToBeAdded },
+      dependencies: { ...untouchedPackageJson.dependencies, ...(await this.addDependencies(packageJsonData.dependencies)) },
+      devDependencies: { ...untouchedPackageJson.devDependencies, ...(await this.addDependencies(packageJsonData.devDependencies)) }
+    };
+
+    fs.writeFileSync(path.resolve(flags.projectDir, 'package.json'), JSON.stringify(adjustedPackageJson, null, 2));
+  }
 
-    fs.writeFileSync(path.resolve(flags.projectDir, 'package.json'), JSON.stringify(packageJson, null, 2));
+  private async addDependencies(dependencies: string[]): Promise<{ [key: string]: string }> {
+    const result: { [key: string]: string } = {};
+    const versionLookups = dependencies.map(dependency => this.getVersionOfDependency(dependency).then(version => (result[dependency] = version)));
+    return Promise.all(versionLookups).then(() => result);
+  }
 
+  private async getVersionOfDependency(dependency: string): Promise<string> {
     try {
-      await execa('npm', ['install', '@sap/cloud-sdk-core'], {
-        cwd: flags.projectDir
-      });
-      await execa('npm', ['install', '--save-dev', '@sap/cloud-sdk-test-util'], { cwd: flags.projectDir });
-    } catch (error) {
-      this.error(error, { exit: 12 });
+      const defaultOptions = ['view', dependency, 'version'];
+      const version = dependency.includes('@sap')
+        ? execa('npm', [...defaultOptions, '--registry', 'https://npm.sap.com'])
+        : execa('npm', defaultOptions);
+
+      return `^${(await version).stdout}`;
+    } catch (err) {
+      this.warn(`Error in finding version for dependency ${dependency} - use LATEST as fallback.`);
+      return 'latest';
     }
   }
 
+  private async installDependencies(flags: Flags) {
+    return execa('npm', ['install'], { cwd: flags.projectDir, stdout: 'inherit' }).catch(err => this.error(`Error in npm install ${err.message}`));
```

Comment:
It might be sensible to add `{ exit: EXITCODE }` param so the program does not finish while also failing to install dependencies (since it could lead to surprising results in a CI/CD context or such)

Reply:
yes